File: CS/PageHitTest/Form1.cs
using System.IO;
using System.Reflection;
using System.Windows.Forms;
using DevExpress.Pdf;
using DevExpress.XtraBars.Ribbon;


namespace PageHitTest {

    public partial class Form1 : RibbonForm {
        public Form1() {
            InitializeComponent();

            Stream stream = GetResourceStream("PageHitTest.demo.pdf");
            pdfViewer.LoadDocument(stream);
        }

        static Stream GetResourceStream(string resourceName) {
            return Assembly.GetExecutingAssembly().GetManifestResourceStream(resourceName);
        }

        void pdfViewer_MouseClick(object sender, MouseEventArgs e) {
            PdfDocumentPosition position = pdfViewer.GetDocumentPosition(e.Location, true);
            MessageBox.Show(string.Format("You clicked on page {0}", position.PageNumber));
        }
    }
}

File: VB/PageHitTest/Form1.vb
Imports System.IO
Imports System.Reflection
Imports System.Windows.Forms
Imports DevExpress.Pdf
Imports DevExpress.XtraBars.Ribbon

Namespace PageHitTest

    Public Partial Class Form1
        Inherits RibbonForm

        Public Sub New()
            InitializeComponent()
            Dim stream As Stream = GetResourceStream("PageHitTest.demo.pdf")
            pdfViewer.LoadDocument(stream)
        End Sub

        Private Shared Function GetResourceStream(ByVal resourceName As String) As Stream
            Return Assembly.GetExecutingAssembly().GetManifestResourceStream(resourceName)
        End Function

        Private Sub pdfViewer_MouseClick(ByVal sender As Object, ByVal e As MouseEventArgs)
            Dim position As PdfDocumentPosition = pdfViewer.GetDocumentPosition(e.Location, True)
            MessageBox.Show(String.Format("You clicked on page {0}", position.PageNumber))
        End Sub
    End Class
End Namespace
